x-pack/plugins/security_solution/public/entity_analytics/components/entity_store/components/dashboard_enablement_panel.tsx

```diff
@@ -0,0 +1,199 @@
+  const entityStoreStatus = state.entityStore.data?.status;
+
+  const [modal, setModalState] = useState({ visible: false });
+  const [riskEngineInitializing, setRiskEngineInitializing] = useState(false);
+
+  const initRiskEngine = useInitRiskEngineMutation();
+  const storeEnablement = useEnableEntityStoreMutation();
+
+  const enableEntityStore = useCallback(
+    (enable: Enablements) => () => {
+      if (enable.riskScore) {
+        const options = {
+          onSuccess: () => {
+            setRiskEngineInitializing(false);
+            if (enable.entityStore) {
+              storeEnablement.mutate();
+            }
+          },
+        };
+        setRiskEngineInitializing(true);
+        initRiskEngine.mutate(undefined, options);
+        setModalState({ visible: false });
+        return;
+      }
+
+      if (enable.entityStore) {
+        storeEnablement.mutate();
+        setModalState({ visible: false });
+      }
+    },
+    [storeEnablement, initRiskEngine]
+  );
+
+  if (storeEnablement.error) {
+    return (
+      <>
+        <EuiCallOut
+          title={
+            <FormattedMessage
+              id="xpack.securitySolution.entityAnalytics.entityStore.enablement.mutation.errorTitle"
+              defaultMessage={'There was a problem initializing the entity store'}
+            />
+          }
+          color="danger"
+          iconType="error"
+        >
+          <p>{(storeEnablement.error as { body: { message: string } }).body.message}</p>
+        </EuiCallOut>
+      </>
+    );
+  }
+
+  if (riskEngineInitializing) {
+    return (
+      <EuiPanel hasBorder data-test-subj="riskEngineInitializingPanel">
+        <EuiEmptyPrompt
+          icon={<EuiLoadingLogo logo="logoElastic" size="xl" />}
+          title={<h2>{ENABLEMENT_INITIALIZING_RISK_ENGINE}</h2>}
+        />
+      </EuiPanel>
+    );
+  }
+
+  if (entityStoreStatus === 'installing' || storeEnablement.isLoading) {
+    return (
+      <EuiPanel hasBorder data-test-subj="entityStoreInitializingPanel">
+        <EuiEmptyPrompt
+          icon={<EuiLoadingLogo logo="logoElastic" size="xl" />}
+          title={<h2>{ENABLEMENT_INITIALIZING_ENTITY_STORE}</h2>}
+          body={
+            <p>
+              <FormattedMessage
+                id="xpack.securitySolution.entityAnalytics.entityStore.enablement.initializing.description"
+                defaultMessage="This can take up to 5 minutes."
+              />
+            </p>
+          }
+        />
+      </EuiPanel>
+    );
+  }
+
+  if (
+    riskEngineStatus !== RiskEngineStatusEnum.NOT_INSTALLED &&
+    (entityStoreStatus === 'running' || entityStoreStatus === 'stopped')
+  ) {
+    return null;
+  }
+
+  const [title, body] = getEnablementTexts(entityStoreStatus, riskEngineStatus);
+  return (
+    <>
+      <EuiEmptyPrompt
+        css={{ minWidth: '100%' }}
+        hasBorder
+        layout="horizontal"
+        className="eui-fullWidth"
+        title={<h2>{title}</h2>}
```

Comment:
I think we should be using EuiTitle here and EuiText the line below?

Reply:
Yeah, i think I've seen that pattern in a few places, but checking the EUI docs [here](https://eui.elastic.co/#/display/empty-prompt), they use as `h2` element...